Add blog post on hardening the Collector (#4759)
open-telemetry/opentelemetry.io: content/en/blog/2024/hardening-the-collector-one.md

---
title: 'Hardening the Collector #1 - A new default bind address'
linkTitle: A new default bind address for the Collector
date: 2024-07-02
author: '[Pablo Baeyens](https://github.com/mx-psi) (OpenTelemetry, Datadog)'
# prettier-ignore

> cSpell:ignore: awsfirehose awsproxy awsxray Baeyens jaegerremotesampling loki remotetap sapm signalfx skywalking splunk

Reviewer: ```suggestion
cSpell:ignore: awsfirehose awsproxy awsxray Baeyens jaegerremotesampling loki remotetap sapm signalfx skywalking splunk
issue: 4760
sig: Collector SIG
```

---

> title: 'Hardening the Collector #1 - A new default bind address'

Reviewer 2: ```suggestion
title: 'Hardening the Collector Episode 1: A new default bind address'
```

_Cue Star Wars opening fanfare_ 🥳

linkTitle: A new default bind address for the Collector
date: 2024-07-02
author: '[Pablo Baeyens](https://github.com/mx-psi) (OpenTelemetry, Datadog)'
# prettier-ignore
cSpell:ignore: awsfirehose awsproxy awsxray Baeyens jaegerremotesampling loki remotetap sapm signalfx skywalking splunk
issue: 4760
sig: Collector SIG
---

The OpenTelemetry Collector recently went through a security audit sponsored by

> the [CNCF](https://www.cncf.io/). As part of this process we recently published

Reviewer 3: Can we use a similar wording from the CVE blog post? "sponsored by, facilitated by, executed by..."?

Author: Done, I used the same wording

a security advisory related to a [DoS vulnerability](/blog/2024/cve-2024-36129/)

> that was fully addressed in v0.102.1.

Reviewer 2: Perhaps a link to the changelog line or PR?

Author: Added a link to the changelog: 0c360aa2b127f2633a9c50de6c506acffc4612c9

The security audit also motivated us to think about ways to harden official
Collector builds and have a more secure default configuration. We are
[actively][releases-586] [working][core-10469] [on several changes][core-10470]
to achieve this and we will be publishing a series of blog posts to keep the

> community informed.

Reviewer 2: As a reader, this leaves me wondering about the baseline or current security status of the Collector. How safe is it? Can we say?

Author: Would it help to link to https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md ? 

Or maybe, the security audit results would also be helpful here. I think we are going to publish them at some point, cc @jpkrohling

Reviewer 3: I would add that those are recommendations, and not classified as issues or vulnerabilities. We can state that we are happy with the results, showing that secure coding practices are a solid part of our routine.

> While we expect the report to be made public soon, we can already say that we are very satisfied with the confirmation that the Collector has proven to be very secure, highlighting the secure coding practices and processes we have in place.

Author: > I would add that those are recommendations, and not classified as issues or vulnerabilities.

I went with 'security best practices'. I also added the sentence you mention :)

- [`jaegerremotesampling` extension](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/jaegerremotesampling#jaegers-remote-sampling-extension)
- [`remotetap` processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/remotetapprocessor#remote-tap-processor)

When in doubt, check the specific components' documentation to see the new
default values.

## What does it mean to me?

If you are relying on the default configuration you may need to start explicitly
setting the endpoint on your Collector components. For example, if you are using
the following configuration with the OTLP receiver:

```yaml
receivers:
  otlp:
    protocols:
      grpc:
```

You may now need to explicitly set the `otlp::protocols::grpc::endpoint`
[configuration setting](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.103.0/receiver/otlpreceiver/config.md):

```yaml
receivers:
  otlp:
    protocols:
      grpc:
        endpoint: ${env:HOST_IP}:4317
```

where the `HOST_IP` environment variable should is set to the bind address you

> want to use (e.g. `status.podIP` on Kubernetes).

Reviewer 2: ```suggestion
want to use (for example, `status.podIP` on Kubernetes).
```

- [`jaegerremotesampling` extension](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/jaegerremotesampling#jaegers-remote-sampling-extension)
- [`remotetap` processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/remotetapprocessor#remote-tap-processor)

When in doubt, check the specific components' documentation to see the new
default values.

## What does it mean to me?

If you are relying on the default configuration you may need to start explicitly
setting the endpoint on your Collector components. For example, if you are using
the following configuration with the OTLP receiver:

```yaml
receivers:
  otlp:
    protocols:
      grpc:
```

You may now need to explicitly set the `otlp::protocols::grpc::endpoint`
[configuration setting](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.103.0/receiver/otlpreceiver/config.md):

```yaml
receivers:
  otlp:
    protocols:
      grpc:
        endpoint: ${env:HOST_IP}:4317
```

where the `HOST_IP` environment variable is set to the bind address you want to
use (for example, `status.podIP` on Kubernetes).

> Note that you are not affected if using the default configuration on the

Reviewer 4: Let's add in the "what we've done" section above when the helm chart changed, and then reference here that you're not affected because "we changed the helm chart a long time ago" or something.

Author: Done! see ecb417d354ce4fdf7429db6e1d05c40ddf6bb314

---
title: 'Hardening the Collector Episode 1: A new default bind address'
linkTitle: A new default bind address for the Collector

> date: 2024-07-02

Reviewer: Let's merge it on the provided date
